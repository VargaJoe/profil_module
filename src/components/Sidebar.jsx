import React, { Component } from 'react';
import Logo from'./Logo';
import Menu from'./Menu';
import User from'./User';

class Sidebar extends Component {
	render () {
		return (
			<div className="sn_sidebar">
				<Logo openMenu={this.props.openMenu}/>
				<User/>
				<Menu />
			</div>
		)
	}
};

export default Sidebar;
