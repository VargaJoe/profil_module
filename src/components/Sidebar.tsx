import * as React			from 'react';
import Logo 				from'./Logo';
import Menu 				from'./Menu';
import User 				from'./User';

export interface MenuTrigger {
	 openMenu
}

class Sidebar extends React.Component<MenuTrigger, {}> {
	
	render () {
		
		return (
			<div className="sn_sidebar">
				<Logo openMenu={ this.props.openMenu } />
				<User/>
				<Menu />
			</div>
		);
	}
};

export default Sidebar;
