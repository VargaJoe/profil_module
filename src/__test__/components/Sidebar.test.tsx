import * as React 						  	from 'react';
import Sidebar 							    from '../../components/Sidebar';
import { 
	configure, 
		shallow,
		mount } 							from 'enzyme';
import * as Adapter 					  	from 'enzyme-adapter-react-16';
import toJson 				      			from 'enzyme-to-json';

configure( {adapter: new Adapter()} );

describe('<NoMatch /> shallow rendering', () => {
		
	const sidebar = shallow(<Sidebar />); 
		
	// test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(sidebar)).toMatchSnapshot();
	});
		
});
