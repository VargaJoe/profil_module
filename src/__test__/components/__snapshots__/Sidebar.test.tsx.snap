// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoMatch /> shallow rendering Match to snapshot 1`] = `
<div
  className="sn_sidebar"
>
  <Logo />
  <Connect(SidebarUser) />
  <Connect(Menu) />
</div>
`;
